jira_scraper/processors/jira_provider.py

```diff
@@ -29,47 +32,54 @@ def __init__(self, query_url: str, query_token: str):
             "Authorization": f"Bearer {query_token}",
         }
 
-    def _get_issues(self, query: str,
-                    max_results: int,
-                    start_at: int = 0) -> str:
+    def get_issues(self, query: str,
+                   max_results: int,
+                   start_at: int = 0) -> tuple[list[dict], int]:
+        """Get issues from Jira.
+
+        Gets issues from Jira and returns list of all the issues and number
+        of retrieved issues.
+
+        Args:
+            query: Query for Jira (e.g., project="ABC")
+            max_results: Maximum number of tickets that should be retrieved
+            start_at: Specififes which chunk of tickets you want to download.
+        """
         full_url = (
             f"{self.query_url}/rest/api/2/search?"
             f"jql={query}&maxResults={max_results}&"
             f"fields=*all&startAt={start_at}"
         )
 
-        response = requests.get(
-            full_url,
-            headers=self.headers,
-            timeout=(3.05, 180),
-        )
-        response.raise_for_status()
-        return json.loads(response.text)
+        LOG.info("Processing Jira request [query: %s, max_results: %d, "
+                 "start_at: %d]", query, max_results, start_at)
 
-    def get_initial_issues(self, query: str,
-                           max_results: int) -> tuple[List[Dict], int]:
-        """Retrieve intial issues and total count."""
         try:
-            data = self._get_issues(query, max_results)
-            return (data["issues"], data['total'])
-        except requests.exceptions.Timeout:
-            print(f"Request to jira query {query} timed out.")
+            response = http.request(
+                method="GET",
+                url=full_url,
+                headers=self.headers,
+                timeout=http.Timeout(connect=3.05, read=180),
+                retries=http.Retry(
```

Comment:
This is fine I guess, another option would have been to use python tenacity: https://tenacity.readthedocs.io

Reply:
I did not know this one. Do you want me to use it, or is keeping it like this ok?

Reply:
no it's alright, it was more for sharing :-)